airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -81,88 +81,42 @@ public ConnectionManagerWorkflowImpl() {}
 
   @Override
   public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws RetryableException {
-    connectionId = connectionUpdaterInput.getConnectionId();
     try {
-      if (connectionUpdaterInput.getWorkflowState() != null) {
-        workflowState = connectionUpdaterInput.getWorkflowState();
-      }
       try {
         syncWorkflowCancellationScope = Workflow.newCancellationScope(() -> {
-          // Scheduling
-          final ScheduleRetrieverInput scheduleRetrieverInput = new ScheduleRetrieverInput(
-              connectionUpdaterInput.getConnectionId());
+          connectionId = connectionUpdaterInput.getConnectionId();
+
+          if (connectionUpdaterInput.getWorkflowState() != null) {
+            workflowState = connectionUpdaterInput.getWorkflowState();
+          }
 
           workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
 
-          final ScheduleRetrieverOutput scheduleRetrieverOutput = configFetchActivity.getTimeToWait(scheduleRetrieverInput);
-          Workflow.await(scheduleRetrieverOutput.getTimeToWait(),
+          Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
```

Comment:
do we still need to wait if we don't want to process the sync?

Reply:
Yes, we don't want to process the sync in 2 different cases: delete and update. If the sync is waiting, it can be manually trigger or launch because of the schedule. so we need to wait and potentially get a signal.

Reply:
got it. can we leave a comment here? it wasn't obvious to me.

Reply:
Done on the `skipScheduling` method.